tools/perf/benchmark_flux.py

```diff
@@ -0,0 +1,153 @@
+parser.add_argument('--inductor', 
+                    action='store_true', 
+                    default=False,
+                    help='Perform benchmark with torch inductor backend (default settings)')
+
+def benchmark_module(pipe, args) -> float:
+    pipe(prompt=args.prompts,
+                 height=args.image_height,
+                 width=args.image_width,
+                 guidance_scale=3.5,
+                 num_inference_steps=args.num_steps,
+                 max_sequence_length=512)
+    
+    torch.cuda.synchronize()
+    start_event = torch.cuda.Event(enable_timing=True)
+    end_event = torch.cuda.Event(enable_timing=True)
+
+    start_event.record()
+    for _ in range(args.iterations):
+        pipe(prompt=args.prompts,
+            height=args.image_height,
+            width=args.image_width,
+            guidance_scale=3.5,
+            num_inference_steps=args.num_steps,
+            max_sequence_length=512)
+    end_event.record()
+    torch.cuda.synchronize()
+
+    return start_event.elapsed_time(end_event) / args.iterations
+
+
+def benchmark_flux_model(args):
+
+    model_names = [] 
+    times = []
+
+    torch_dtype = torch.float32
+    options = {}
+    if args.bf16:
+        options["bf16"] = True
+        torch_dtype = torch.bfloat16
+
+    pipe = FluxPipeline.from_pretrained("black-forest-labs/FLUX.1-dev", torch_dtype=torch_dtype)
+    pipe = pipe.to("cuda")
+
+    torch_res = benchmark_module(pipe, args)
+    model_names.append("Torch Model")
+    times.append(torch_res)
+
+    del pipe 
+
+    if "migraphx" in torch._dynamo.list_backends():
+        torch._dynamo.reset()
+
+        pipe = FluxPipeline.from_pretrained("black-forest-labs/FLUX.1-dev", torch_dtype=torch.float32)
+        pipe = pipe.to("cuda")
+
+        pipe.text_encoder = torch.compile(pipe.text_encoder, backend='migraphx', options=options)
+        pipe.text_encoder_2 = torch.compile(pipe.text_encoder_2, backend='migraphx', options=options)
+        pipe.transformer = torch.compile(pipe.transformer, backend='migraphx', options=options)
+        pipe.vae.decoder = torch.compile(pipe.vae.decoder, backend='migraphx', options=options)
+
+        mgx_dynamo_res = benchmark_module(pipe, args)
+        
+        model_names.append("MIGraphX Dynamo")
+        times.append(mgx_dynamo_res)
+        del pipe
+
+
+    if args.inductor:
+        torch._dynamo.reset()
+        
+        pipe = FluxPipeline.from_pretrained("black-forest-labs/FLUX.1-dev", torch_dtype=torch_dtype)
+        pipe = pipe.to("cuda")
+
+        pipe.text_encoder = torch.compile(pipe.text_encoder)
+        pipe.text_encoder_2 = torch.compile(pipe.text_encoder_2)
+        pipe.transformer = torch.compile(pipe.transformer)
+        pipe.vae.decoder = torch.compile(pipe.vae.decoder)
+
+        inductor_res = benchmark_module(pipe, args)
+
+        model_names.append("Torch Inductor")
+        times.append(inductor_res)
+        del pipe
+
+    print_bm_results(model_names, times, 1)
```

Comment:
```suggestion
    print_bm_results(model_names, times, 0)
```
should compare the times to eager timing (idx 0) to measure speedup i think

Reply:
ahhh oops